bundles/ch.elexis.core.mail.ui/src/ch/elexis/core/mail/ui/dialogs/SendMailDialog.java

```diff
@@ -270,7 +270,11 @@ public void selectionChanged(SelectionChangedEvent event) {
 					updateLayout();
 				}
 			});
-
+			if ("true".equals(getOkLabel())) {
```

Comment:
Das könnte auch ohne neue String Konstante mit `Boolean.valueOf(getOkLabel())` gelöst werden.

Wieso wird hier von getOkLabel true erwartet? Sollte da nicht jeglicher String möglich sein? Und nur wenn da irgendwas gesetzt ist, das den Effekt haben das der Button den entsprechenden String als Label bekommt, und nicht sofort gesendet wird?

Reply:
Du hast vollkommen recht. Ich habe fälschlicherweise angenommen, dass getOkLabel immer "true" zurückgeben sollte, weil ich es im EmailEditHandler so übergebe. Aber ja, es sollte in der Tat jeglichen String annehmen können. Dein Vorschlag mit Boolean.valueOf(getOkLabel()) ist sinnvoll. Wenn ein Wert für getOkLabel gesetzt ist, sollte der Button das entsprechende Label erhalten und nicht sofort gesendet werden. Vielen Dank für deinen Input